pypesto/petab/importer.py

```diff
@@ -78,6 +80,38 @@ def from_yaml(yaml_config: Union[dict, str],
             output_folder=output_folder,
             model_name=model_name)
 
+    def check_and_log_gradients(self, RTOL: float = 1e-2, ATOL: float = 1e-3):
+        """Check and log if gradients are compatible with gradient-based optimizers
+        Parameters
+        ----------
+        RTOL: relative error tolerance (default 1e-2)
+        ATOL: absolute error tolerance (default 1e-3)
+        """
+        par = np.asarray(self.petab_problem.x_nominal_scaled)
+        problem = self.create_problem()
+        objective = problem.objective
+        objective.amici_solver.setSensitivityMethod(
+            amici.SensitivityMethod_forward)
+        objective.amici_solver.setAbsoluteTolerance(1e-10)
+        objective.amici_solver.setRelativeTolerance(1e-12)
+        free_indices = par[problem.x_free_indices]
+        dfs = []
+
+        modes = [MODE_FUN, MODE_RES]
+        for mode in modes:
```

Comment:
This loop over `modes` could be by default, but also allow users to only specify one of `MODE_FUN` or `MODE_RES`, to reduce computation (perhaps not a significant issue except for large problems).

Reply:
Okay, default is to consider both modes but give the user control over it.